src/main/scala/uk/gov/hmrc/scalatestaccessibilitylinter/AccessibilityLinterMatchers.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.scalatestaccessibilitylinter
+
+import org.scalatest.matchers.{MatchResult, Matcher}
+import play.api.libs.json._
+import scala.collection.immutable.ListMap
+
+trait AccessibilityLinterMatchers {
+  protected val accessibilityLinters: Seq[AccessibilityLinter]
+
+  lazy val passAccessibilityChecks = new PassAccessibilityChecksMatcher(accessibilityLinters)
+}
+
+class PassAccessibilityChecksMatcher(accessibilityLinters: Seq[AccessibilityLinter]) extends Matcher[String] {
+  def apply(html: String): MatchResult = {
+    val accessibilityLinterReports = accessibilityLinters.map(_.check(html))
+
+    val checksPassed = accessibilityLinterReports.forall(_.noUnknownIssues)
+
+    MatchResult(
+      checksPassed,
+      s"Accessibility violations were present...${reportAsJsonLines(accessibilityLinterReports)}",
+      s"Accessibility violations were not present..."
+    )
+  }
+
+  private def reportAsJsonLines(reports: Seq[AccessibilityReport]): String =
+    reports
+      .collect({ case FailedAccessibilityChecks(violations) => violations })
+      .flatten
+      .map { v =>
+        Json.toJson(
+          ListMap(
+            "level"       -> (if (v.knownIssue) "WARN" else v.alertLevel),
+            "description" -> v.description,
+            "snippet"     -> v.snippet,
+            "helpUrl"     -> v.helpUrl,
+            "furtherInfo" -> v.furtherInformation.getOrElse("")
+          )
+        )
```

Comment:
probably better done as a Json Writes thing on AccessibilityViolation, but for now I think leave it here so it's easy to change it just for test output if we want